### ISSUE TITLE
fenêtre Projet

### ISSUE THREAD

**Issue by marcisenmann** [cbn-alpin/gefiproj-webapp#3]
- Il n'est plus possible de créer de projet. Peut être, lié à la MAJ sur le format de l'ID du projet à 5 chiffres ? Le message indique "que le code projet doit comprendre 4 chiffres". A voir

**Comment by manu0005** [comment]
Oui, nous n'avons pas encore mis à niveau la totalité des serveurs (Back, Front et BDD) avec les dernières modifications...
Nous essayons de livrer régulièrement nos travaux sur la machine OVH, mais il peut y a avoir une discordance temporaire entre le Back-end, le Front-end ou même la BDD (car nous l'utilisons aussi pour nos développements).

**Comment by marcisenmann** [comment]
Je m'en doutais. Pas de soucis.
à demain,
Marc

**Comment by marcisenmann** [comment]
Memo/retours :
- à l'ouverture de Projets : tri des projets sur le code par ordre décroissant ? soit les codes les plus récents en haut.

**Comment by marcisenmann** [comment]
- Préciser les modalités de suppression d'un projet + autres données en cascade, Si ce cas peut arriver évidemment (à voir avec Bertrand).

**Comment by marcisenmann** [comment]
- Peut-on solder un projet sans qu'il y ait de [Date arrêté ou commande] ? voir avec Bertrand si ce cas de figure peut se rencontrer (ex : un projet qui tombe à l'eau avec promesse de financement annulée...)

**Comment by manu0005** [comment]
Ce jour, Bertrand nous a donné un certain nombre de réponses : 
- Bertrand est habitué à avoir un trie décroissant sur le code des projets.
- effectivement un projet peut-être soldé alors qu'il n'a pas de date 'arrêté ou commande'
- "Préciser les modalités de suppression d'un projet + autres données en cascade" ça n'a pas été évoqué il me semble. Nous ne permettons pas de supprimer un projet ayant des financements (cela avait été acté précédemment), par contre nous supprimons en cascades les éléments enfants (recettes et affectations) sur la suppression d'un financement ou d'une recette. A voir si c'est bien le comportement désiré. Voilà la boîte de dialogue de mise en garde :
![image](https://user-images.githubusercontent.com/49067444/107259655-8d3bfc00-6a3d-11eb-8de8-842d71c9d66b.png)

**Comment by marcisenmann** [comment]
Oui, il faudra confirmer avec Bertrand les règles en question. Elles peuvent très bien convenir en l'état pour la suppression d'un financement. Pour ce qui concerne la suppression d'un projet, ça vaut le coup d'en rediscuter rapidement, simplement pour préciser si cela peut être nécessaire ou pas. Bonne soirée.

**Comment by marcisenmann** [comment]
Je pense à des projets qui ne se concrétisent pas. Les mettre en statut de projet soldé ne serait pas bien adapté je pense. À voir !

**Comment by manu0005** [comment]
Je parlerai de cela à Jean-Pascal cet après-midi. En attendant je vais créer une entrée dans le Wiki/RG pour indiquer les éléments à définir.

**Comment by marcisenmann** [comment]
Le déroulement en place peut en fait très bien convenir. J'ai fait un test sur un de mes projets l'autre jour : si tu supprimes les éléments enfants les uns après les, autres (montants, recettes puis financement), tu peux supprimer au final un projet. Ca peut rester comme cela.

**Comment by manu0005** [comment]
Parfait. Je retire donc l'entrée dans le Wiki. :)
